spec/bolt/cli_spec.rb

```diff
@@ -89,6 +89,167 @@ def stub_config(file_content = {})
+
+      expect { cli.parse }.to raise_error(
+        Bolt::CLIError,
+        /Must specify a plan name/
+      )
+    end
+
+    it 'calls #new_plan' do
+      allow(cli).to receive(:new_plan).and_return(0)
+      expect(cli).to receive(:new_plan).with(plan_name)
+      cli.execute(cli.parse)
+    end
+
+    describe '#new_plan' do
+      it 'errors without a named project' do
+        allow(project).to receive(:name).and_return(nil)
+        cli.parse
+
+        expect { cli.new_plan(plan_name) }.to raise_error(
+          Bolt::Error,
+          /Project directory '.*' is not a named project/
+        )
+      end
+
+      it 'errors when the plan name is invalid' do
+        cli.parse
+
+        %w[Foo foo-bar foo:: foo::Bar foo::1bar ::foo].each do |plan_name|
+          expect { cli.new_plan(plan_name) }.to raise_error(
+            Bolt::ValidationError,
+            /Invalid plan name '#{plan_name}'/
+          )
+        end
+      end
+
+      it 'errors if the first name segment is not the project name' do
+        plan_name = 'plan'
+        cli.parse
+
+        expect { cli.new_plan(plan_name) }.to raise_error(
+          Bolt::ValidationError,
+          /First segment of plan name '#{plan_name}' must match project name/
+        )
+      end
+
+      %w[pp yaml].each do |ext|
+        it "errors if there is an existing #{ext} plan with the same name" do
+          plan_path = File.join(project_path, 'plans', "init.#{ext}")
+          FileUtils.mkdir(File.dirname(plan_path))
+          FileUtils.touch(plan_path)
+
+          cli.parse
+
+          expect { cli.new_plan(plan_name) }.to raise_error(
+            Bolt::Error,
+            /A plan with the name '#{plan_name}' already exists/
+          )
+        end
+      end
+
+      it "creates a missing 'plans' directory" do
+        cli.parse
+        expect(Dir.exist?(project.plans_path)).to eq(false)
+        cli.new_plan(plan_name)
+        expect(Dir.exist?(project.plans_path)).to eq(true)
+      end
+
+      it 'creates a missing directory structure' do
+        plan_name = "#{project_name}::foo::bar"
+        cli.parse
+        expect(Dir.exist?(project.plans_path + 'foo')).to eq(false)
+        cli.new_plan(plan_name)
+        expect(Dir.exist?(project.plans_path + 'foo')).to eq(true)
+      end
+
+      it 'catches existing file errors when creating directories' do
+        plan_name = "#{project_name}::foo::bar"
+        FileUtils.mkdir(File.join(project_path, 'plans'))
+        FileUtils.touch(File.join(project_path, 'plans', 'foo'))
+
+        cli.parse
+
+        expect { cli.new_plan(plan_name) }.to raise_error(
+          Bolt::Error,
+          /unable to create plan directory/
+        )
+      end
+
+      it "creates an 'init' plan when the plan name matches the project name" do
+        cli.parse
+        cli.new_plan(plan_name)
+
+        plan_path = project.plans_path + 'init.yaml'
+
+        expect(File.size?(plan_path)).to be
```

Comment:
Can we just check the file exists?

Reply:
The intent was to check that it exists and has content written to it, but I can change it to a simpler `File.exist?`.